build.gradle

```diff
@@ -2,91 +2,99 @@ import se.bjurr.gitchangelog.plugin.gradle.GitChangelogTask
 	configurations.classpath {
 		resolutionStrategy {
-			cacheDynamicVersionsFor 10, "seconds"
-			cacheChangingModulesFor 0, "seconds"
+			cacheDynamicVersionsFor 10, 'seconds'
+			cacheChangingModulesFor 0, 'seconds'
 		}
 	}
 	dependencies {
-		classpath group: "net.minecraftforge.gradle", name: "ForgeGradle", version: "3.+", changing: true
+		classpath group: 'net.minecraftforge.gradle', name: 'ForgeGradle', version: '4.1.+', changing: true
 	}
 }
 
 plugins {
-	id "com.matthewprenger.cursegradle" version "1.4.0"
-	id "se.bjurr.gitchangelog.git-changelog-gradle-plugin" version "1.64"
+	id 'java'
+	id 'eclipse'
+	id 'idea'
+	id 'com.matthewprenger.cursegradle' version '1.4.0'
+	id 'se.bjurr.gitchangelog.git-changelog-gradle-plugin' version '1.65'
 }
-apply from: "buildtools/ColoredOutput.gradle"
-apply plugin: 'idea'
-apply plugin: "net.minecraftforge.gradle"
+apply plugin: 'net.minecraftforge.gradle'
+apply from: 'buildtools/ColoredOutput.gradle'
 
 //adds the build number to the end of the version string if on a build server
 String build_number = System.getenv().BUILD_NUMBER
 if (build_number == null) {
-	build_number = "9999"
+	build_number = '9999'
 }
 
 String semver = "${version_major}.${version_minor}.${version_patch}"
 version = "${semver}.${build_number}"
-group = "mezz.jei" // http://maven.apache.org/guides/mini/guide-naming-conventions.html
-
-sourceCompatibility = targetCompatibility = JavaVersion.VERSION_1_8
-compileJava {
-	sourceCompatibility = targetCompatibility = JavaVersion.VERSION_1_8
-}
-
-repositories {
-	maven { url 'https://repo1.maven.org/maven2' }
-	maven { url 'https://maven.latmod.com/' }
-}
+group = 'mezz.jei' // http://maven.apache.org/guides/mini/guide-naming-conventions.html
 
 sourceSets {
 	api
-	main
-	test
+	main {
+		compileClasspath += sourceSets.api.output
+		runtimeClasspath += sourceSets.api.output
+		resources {
+			srcDir 'src/generated/resources'
+		}
+	}
+	test {
+		compileClasspath += sourceSets.api.output
+		runtimeClasspath += sourceSets.api.output
+	}
 }
 
 configurations {
-	apiCompile.extendsFrom(compile)
+	apiImplementation.extendsFrom(implementation)
+	apiRuntimeOnly.extendsFrom(runtimeOnly)
+}
+
+sourceCompatibility = targetCompatibility = compileJava.sourceCompatibility = compileJava.targetCompatibility = JavaVersion.VERSION_1_8
+
+repositories {
+	mavenCentral()
+	maven { url 'https://maven.latmod.com/' }
 }
 
 dependencies {
 	minecraft([
-		group  : "${project.forgegroup}",
-		name   : "forge",
-		version: "${project.mcversion}-${project.forgeversion}"
+			group  : "${project.forgegroup}",
+			name   : 'forge',
+			version: "${project.mcversion}-${project.forgeversion}"
 	])
-	implementation sourceSets.api.output
-	testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.2'
-	testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.6.2'
+	testImplementation platform('org.junit:junit-bom:5.7.1')
+	testImplementation 'org.junit.jupiter:junit-jupiter'
```

Comment:
This seems like an odd change

Reply:
There are two possibilities and JUnit suggests both...
- https://junit.org/junit5/docs/current/user-guide/#running-tests-build-gradle-engines-configure
- https://github.com/junit-team/junit5-samples/blob/r5.7.1/junit5-jupiter-starter-gradle-groovy/build.gradle

Reply:
I prefer keeping the version on both